src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -289,6 +289,39 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to target fields types only:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name=&quot;query&quot; value=&quot;//METHOD_DEF//*&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * import java.util.Optional;
+ *
+ * public class Main {
+ *
+ *   static int field1 = 4; // OK
+ *   public Optional&lt;String&gt; field2; // violation, usage of type 'Optional' is not allowed
+ *   protected String field3; // OK
+ *   Optional&lt;String&gt; field4; // violation, usage of type 'Optional' is not allowed
+ *   private Optional&lt;String&gt; field5; // violation, usage of type 'Optional' is not allowed
+ *
+ *   void foo() {
+ *     Optional&lt;String&gt; i; // OK
+ *   }
+ *   public &lt;T extends java.util.Optional&gt; void method(T t, Optional&lt;T&gt; j) { // OK
```

Comment:
Please change code back to what it was, leave `// OK` comment.  We just want to show that the bounded type parameter, `<T extends java.util.Optional>` is `// OK` too, since IllegalType checks this.

Reply:
done

Reply:
done

Reply:
Resolved.

Reply:
i think we should not have revert the changes here 
see when we use this `<property name="tokens" value="PARAMETER_DEF"/>` with 
` *   public &lt;T extends java.util.Optional&gt; void method(T t, Optional&lt;T&gt; j) { // OK
`
then we would get violation but is we remove `, Optional&lt;T&gt; j` then we won't

Reply:
i think we should not revert the changes here 
see when we use this `<property name="tokens" value="PARAMETER_DEF"/>` with 
` *   public &lt;T extends java.util.Optional&gt; void method(T t, Optional&lt;T&gt; j) { // OK
`
then we would get violation but is we remove `, Optional&lt;T&gt; j` then we won't 
also there is no example above similar to this

Reply:
> see when we use this `<property name="tokens" value="PARAMETER_DEF"/>`

Please explain why we should change the tokens.  The issue that we are addressing in this PR is to show a very specific use case of this check where user wants to target ONLY fields.

> also there is no example above similar to this

What is this comment referring to?

Reply:
no i am not saying we change the config it was just to show the need of `, Optional&lt;T&gt; j` 
and `What is this comment referring to?` 
https://checkstyle.org/config_coding.html#IllegalType_Examples if u see here you will find that with `PARAMETER_DEF` there is no example or with this `public <T extends java.util.Optional> void method(T t, Optional<T> j)`

Reply:
no i am not saying we change the config it was just to show the need of `, Optional&lt;T&gt; j` 
and `What is this comment referring to?` 
https://checkstyle.org/config_coding.html#IllegalType_Examples if u see here you will find that with `PARAMETER_DEF` there is no example or with this           
`public <T extends java.util.Optional> void method(T t, Optional<T> j)`

Reply:
If you would like to show that method parameters definitions are `// OK`, that is a good idea, but it would be better to do that in the first method so that we can see each `// OK` on a separate line.  Otherwise, I think it would require an explanation.